tests/test_requests.py

```diff
@@ -1026,13 +1026,13 @@ def test_response_reason_unicode_fallback(self):
         # check raise_status falls back to ISO-8859-1
         r = requests.Response()
         r.url = 'some url'
-        reason = b'Komponenttia ei l\xf6ydy'
-        r.reason = reason
+        reason = u'Komponenttia ei löydy'
+        r.reason = reason.encode('latin-1')
         r.status_code = 500
         r.encoding = None
         with pytest.raises(requests.exceptions.HTTPError) as e:
             r.raise_for_status()
-        assert reason.decode('latin-1') in str(e)
+        assert reason in e.value.args[0]
```

Comment:
e.value.args[0] is the message which is unicode. I believe this is part of the problem I'm discussing in the second paragraph.

```python
import pytest
import requests

r = requests.Response()
r.url = 'some url'
reason = u'Komponenttia ei löydy'
r.reason = reason.encode('latin-1')
r.status_code = 500
r.encoding = None
with pytest.raises(requests.exceptions.HTTPError) as e:
        r.raise_for_status()
assert reason in e.value.args[0]
assert isinstance(e.value.args[0], unicode)
```

Note that `e` is pytest's `ExceptionInfo` class not an `HTTPError` in this case. That's why we're accessing `e.value.args[0]` instead of `e.args[0]`. `e.message` doesn't work in Python 3.